editor/src/components/shared/project-components.ts

```diff
@@ -294,10 +294,7 @@ export function getComponentGroups(
         file.fileContents.parsed,
       )
       if (possibleExportedComponents != null) {
-        const componentsWithoutApp_KILLME = possibleExportedComponents.filter(
```

Comment:
I think the depth detection is a good idea, but I'd keep the filtering out of App, as a fix for an obvious default project footgun.

Reply:
I don't think that is a call we should make tbh. The default standard is to use `App` as the main component, but that's not something which is set in stone or required at all. As long as this PR catches that case I'd say that's good enough.

Reply:
well, we will let you insert App and immediately show an exception screen on the canvas, making you undo what you just did. I'd say for the specific case of App, that's a regression in user experience

Reply:
Well you could argue the inverse, as it was preventing anything from being inserted if it was called `App`.